interchain-exchange/tutorial/03-orderbook.md

```diff
@@ -121,36 +162,40 @@ func AppendOrder(book OrderBook, creator string, amount int32, price int32) (Ord
 	order.Price = price
 
 	// Increment ID tracker
-	book = book.IncrementNextOrderID()
+	book.IncrementNextOrderID()
 
 	// Insert the order
-	book = book.InsertOrder(order)
+	book.insertOrder(order, ordering)
 
-	return book, order.Id, nil
+	return order.Id, nil
 }
 
-// checkAmountAndPrice checks correct amount or price
-func checkAmountAndPrice(amount int32, price int32) error {
-	if amount == int32(0) {
-		return ErrZeroAmount
-	}
-	if amount > MaxAmount {
-		return ErrMaxAmount
-	}
-	if price == int32(0) {
-		return ErrZeroPrice
-	}
-	if price > MaxPrice {
-		return ErrMaxPrice
-	}
+// insertOrder inserts the order in the book with the provided order
+func (book *OrderBook) insertOrder(order Order, ordering Ordering) {
+	if len(book.Orders) > 0 {
+		var i int
 
-	return nil
+		// get the index of the new order depending on the provided ordering
+		if ordering == Increasing {
+			i = sort.Search(len(book.Orders), func(i int) bool { return book.Orders[i].Price > order.Price })
+		} else {
+			i = sort.Search(len(book.Orders), func(i int) bool { return book.Orders[i].Price < order.Price })
+		}
+
+		// insert order
+		orders := append(book.Orders, &order)
+		copy(orders[i+1:], orders[i:])
+		orders[i] = &order
+		book.Orders = orders
+	} else {
+		book.Orders = append(book.Orders, &order)
+	}
 }
 ```
 
 ## Add The Sellorder
 
-Modify the `sellOrderBook.proto` file to have the fields for creating a sell order on the order book.
+Modify the `sellOrderBook.proto` file to add the order book into the sell order book.
```

Comment:
Modify the `sell_order_book.proto` file to add the order book into the sell order book.

Reply:
Comment on a previous version of the tutorial.